pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
+		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
+		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
```

Comment:
So this concerns me because for custom deployments it's almost certainly
going to be wrong.  I'm worried about breaking by default.  I would prefer
we set requests.

On Thu, Sep 1, 2016 at 10:59 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/deploy/controller/deployment/controller.go
> <https://github.com/openshift/origin/pull/10761#discussion_r77190686>:
>
> > @@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
> >
> >  	gracePeriod := int64(10)
> >
> > +	// In case the deployment config strategy does not set the resource requests,
> > +	// we default them.
> > +	if deploymentConfig.Spec.Strategy.Resources.Limits == nil {
> > +		deploymentConfig.Spec.Strategy.Resources.Limits = kapi.ResourceList{}
> > +		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceCPU] = resource.MustParse("100m")
> > +		deploymentConfig.Spec.Strategy.Resources.Limits[kapi.ResourceMemory] = resource.MustParse("256Mi")
>
> @smarterclayton <https://github.com/smarterclayton> I know we defaulting
> this for registry and router, just want to double check if these values are
> correct?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10761/files/e16582ed14cbad28129667f6726e0926f3b8b204#r77190686>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6dhKdciZ11_D32J-YEGawrDG4eOks5qluhIgaJpZM4JypmU>
> .
>